packages/signals/browser-signals/src/core/signal-generators/network.ts

```diff
@@ -0,0 +1,98 @@
+import { SignalEmitter } from '../emitter'
+import { SignalGenerator } from './types'
+let origFetch: typeof window.fetch
+
+export function addFetchInterceptor(
+  onRequest: (rs: any) => void,
+  onResponse: (rs: Response) => void
+) {
+  console.debug('Adding fetch interceptor')
+  origFetch = window.fetch
+  window.fetch = async (...args) => {
+    try {
+      onRequest(args)
+    } catch (err) {
+      console.log('Error handling request: ', err)
+    }
+    const response = await origFetch(...args)
+    try {
+      onResponse(response.clone())
+    } catch (err) {
+      console.log('Error handling response: ', err)
+    }
+    return response
+  }
+}
+
+const normalizeHeaders = (headers: HeadersInit): Headers => {
+  return headers instanceof Headers ? headers : new Headers(headers)
+}
+
+const containsJSONContent = (headers: HeadersInit | undefined): boolean => {
+  if (!headers) {
+    return false
+  }
+  const normalizedHeaders = normalizeHeaders(headers)
+  return (
+    normalizedHeaders.get('content-type')?.includes('application/json') ||
+    normalizedHeaders.get('Content-Type')?.includes('application/json') ||
```

Comment:
Headers.get is case insensitive, you don't need to check for content-type or Content-type -- just `content-type` is sufficient (maybe right a test to the affect, but I just tested) 😄